### PR TITLE
translate-c: RemAssign and DivAssign

### LINKED ISSUE #4790: Teach translate-c to handle RemAssign
A good and simple first task.

```C
int foo() {
    int x = 10;
    x %= 3;
    return x;
}
```

### PR DESCRIPTION
**Pull request by phase**
Closes #4790

Let me know if there is anything else I should do!

### CODE REVIEW

File: src-self-hosted/translate_c.zig [hunk 1]
```diff
@@ -3018,6 +3018,8 @@ fn transCompoundAssignOperator(rp: RestorePoint, scope: *Scope, stmt: *const Zig
             return transCreateCompoundAssign(rp, scope, stmt, .AssignSubWrap, .MinusPercentEqual, "-%=", .SubWrap, .MinusPercent, "-%", used)
         else
             return transCreateCompoundAssign(rp, scope, stmt, .AssignSub, .MinusPercentEqual, "-=", .Sub, .Minus, "-", used),
+        .DivAssign => return transCreateCompoundAssign(rp, scope, stmt, .AssignDiv, .SlashEqual, "/=", .Div, .Slash, "/", used),
+        .RemAssign => return transCreateCompoundAssign(rp, scope, stmt, .AssignMod, .PercentEqual, "%=", .Mod, .Percent, "%", used),
```
Comment (Vexu) [inline src-self-hosted/translate_c.zig:3022]: Signed integer divisions have to use `@divTrunc` and `@rem` builtin functions. Take a look at `transBinaryOperator` for help on doing the checks.
Also it seems that `transBinaryOperator` is checking `!cIsUnsignedInteger(qt)` when it should be checking `cIsSignedInteger(qt)` so you could also fix that.

Comment (phase) [reply src-self-hosted/translate_c.zig:3022]: Ah, I'll switch it to use those!

**Comment by phase** [comment]
The new test cases should be generating the proper code with `@divTrunc` and `@rem` for signed integers and `/` & `%` for unsigned integers.

File: src-self-hosted/translate_c.zig [hunk 2]
```diff
@@ -3095,31 +3121,53 @@ fn transCreateCompoundAssign(
...
+        try block_scope.block_node.statements.push(assign);
+    }
 
     const break_node = try transCreateNodeBreak(rp.c, block_scope.label);
     break_node.rhs = ref_node;
     try block_scope.block_node.statements.push(&break_node.base);
     block_scope.block_node.rbrace = try appendToken(rp.c, .RBrace, "}");
     // semicolon must immediately follow rbrace because it is the last token in a block
-    _ = try appendToken(rp.c, .Semicolon, ";");
+    _ = try appendToken(rp.c, .Semicolon, ";4");
```
Comment (Vexu) [inline src-self-hosted/translate_c.zig:3170]: This semicolon can be removed since it is not needed inside a grouped expression. Otherwise LGTM.

Comment (daurnimator) [inline src-self-hosted/translate_c.zig:3170]: accidental `4`?

Comment (phase) [reply src-self-hosted/translate_c.zig:3170]: Oops I forgot to remove my testing 😅 I'll get that fixed up.

**Pull request merged**